### PR TITLE
build adserver dockerimage using poetry-defined environment

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 6f3aeedc3b0f89a4804156a0d0c28748d35eb52a | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3783-integration-pqhp2) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | daba979176e2d4778e975ca5d7245b1a179683b8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3783-integration-78xt6) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: components/alibi-explain-server/Dockerfile [hunk 1]
```diff
@@ -60,6 +60,7 @@ RUN yum -y update-minimal --security --sec-severity=Important --sec-severity=Cri
 RUN yum remove -y nodejs httpd
 
 COPY --from=builder /microservice /microservice
+COPY --from=builder /licenses /licenses
```
Comment (axsaucedo) [inline components/alibi-explain-server/Dockerfile:63]: Were the licenses not generated before? Normally we do have the requirement to include all licenses inside the image, but I'm assuming AD never actually had the scipts to extract all the licenses

Comment (RafalSkolasinski) [reply components/alibi-explain-server/Dockerfile:63]: It was missed when introduced double-stage build. That happened in master after 1.11.2 and it was just a mistake when introducing poetry in alibi explain server that I spotted now when doing detect server. 

So we always had these in what we released, master was missing it for some time, and this fixes it.

File: examples/feedback/metrics-server/config/multiclass-deployment.yaml [hunk 2]
```diff
@@ -43,7 +43,7 @@ spec:
         - name: "PREDICTIVE_UNIT_ID"
           value: "classifier"
         - name: "PREDICTIVE_UNIT_IMAGE"
-          value: "alibi-detect-server:1.7.0-dev"
+          value: "seldonio/alibi-detect-server:1.12.0-dev"
```
Comment (axsaucedo) [inline examples/feedback/metrics-server/config/multiclass-deployment.yaml:46]: Was this one teh one that was added to the release.py

Comment (RafalSkolasinski) [reply examples/feedback/metrics-server/config/multiclass-deployment.yaml:46]: I still need to add it to release.py (or check if maybe with `seldonio/` it is caught now)

Comment (RafalSkolasinski) [reply examples/feedback/metrics-server/config/multiclass-deployment.yaml:46]: But yeah, that was the one we talked about  that it stayed on 1.7.0-dev

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3783#issuecomment-988745792 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | daba979176e2d4778e975ca5d7245b1a179683b8 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3783-integration-78xt6) | `/test integration`
notebooks | 56ca263acbbb5cc4130459220ce008b5f86547bd | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3783-notebooks-89z4t) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 56ca263acbbb5cc4130459220ce008b5f86547bd | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-3783-notebooks-89z4t) | `/test notebooks`
integration | 56ca263acbbb5cc4130459220ce008b5f86547bd | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3783-integration-5j4t4) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Just for records:


Integration failures on:
- TestRollingHttp.test_rolling_update5[ambas]
- TestRollingHttp.test_rolling_update10[ambas]
- test_label_update[1.7.0]
- test_label_update[1.10.0]
- TestADServer.test_alibi_detect_cifar10_rclone

The rclone test is fixed in followup [PR](https://github.com/SeldonIO/seldon-core/pull/3796).

Notebooks failures on:
- TestNotebooks.test_upgrade
- TestNotebooks.test_server_examples

**Pull request merged**